### ISSUE TITLE
[Feature]: Show the colour picker modal for additional file extensions

### ISSUE THREAD

**Issue by NashJames** [microsoft/vscode#187395]
As far as I can find, the colour picker and small colour preview mainly appears for files with extensions such as `.css`, `.scss` and `.tsx`. It will not appear for files like `.ts`. An example of this being extremely useful is a tailwind config file where switching to a `.tsx` extension can create compiler errors. Defining theme colours in non-XML typescript can be annoying to refer back to.

The simplest solution might be a json preferences property to allow us to specify a regex of file extensions we wish for it to be visible on. This would allow us to add something like `"editor.colorPicker": ["*.coffee", "./src/lib/tailwind-theme.ts"]` so we can reference specific files or an entire extension.

I'd prefer not to install more and more extensions, especially where the feature already exists. Hopefully this one is simple since it only needs a config option 😃

**Comment by NashJames** [comment]
Hey! Thanks for the reply. This one:

>  the color picker which appears when you hover over a color decorator (colored box in the editor)

So as you reference it, I'd like to specify some additional files for color decorators to appear in

**Comment by NashJames** [comment]
Okay interesting. I never found  `editor.defaultColorDecorators`, even when having a look through the official docs.

Now that I'm aware what the problematic vscode setting is, I ended up searching again and found #182214. Similarly for me `editor.defaultColorDecorators` did not work until I removed the tailwind extension. Seems a very odd mechanic to automatically overwrite useful features.

Hopefully now with my upvote bringing the total to 20, that issue should now be in the backlog to be improved.

Thanks @aiday-mar. Looking forward to seeing it resolved.

Closing.
